kewl-kafkacluster-processor/src/main/scala/com/mwam/kafkakewl/processor/kafkacluster/KafkaClusterAdmin.scala

```diff
@@ -153,8 +169,37 @@ private[kafkacluster] class DefaultKafkaClusterAdmin(
     change match {
       case KafkaClusterChange.Add(topic: KafkaClusterItem.Topic) =>
         assert(topic.isReal, "cannot deploy (add) un-real topic")
-        ifNotDryRun {
-          adminClient.createTopic(topic.name, topic.partitions, topic.replicationFactor, topic.config)
+        for { // based on Try
+          deleteTopicsCheckResult <- Try {
+            // If there is no zookeeper configured for this cluster, we just pretend that there is nothing in '/admin/delete_topics'
+            Try(zkClientOption.map(_.getChildren.forPath("/admin/delete_topics").asScala.toList).getOrElse(Nil)) match {
+              case Failure(t) =>
+                val errorMessage = s"could not check the '/admin/delete_topics' node in ZK before creating the topic '${topic.name}': ${t.getMessage}"
+                // I want to log this as an error with a specific message here (it'll be logged later as part of the deployment result too, but it's easier to find it if we log as a separate error)
+                logger.error(errorMessage)
+                metrics.counter(s"zk_get_deleted_topics_failed:$kafkaClusterId").inc()
+                // We go ahead with the topic creation, it's probably better than not even trying. I'm not sure how often and when exactly
+                // we can get into this code branch, we'll see if this needs to change.
+                KafkaOperationResult.tell(errorMessage)
+              case Success(deleteTopics) =>
+                if (deleteTopics.contains(topic.name)) {
+                  val exception = new KafkaTopicPendingDeleted(topic.name)
+                  // I want to log this as an error with a specific message here (it'll be logged later as part of the deployment result too, but it's easier to find it if we log as a separate error)
+                  logger.error(exception.getMessage)
+                  metrics.counter(s"not_created_pending_deleted_topic:$kafkaClusterId:${topic.name}").inc()
+                  // the outer Try ensures that we fail-fast before attempting to create the topic
+                  throw exception
+                } else {
+                  KafkaOperationResult.empty
+                }
+            }
+          }
+          createTopicResult <- ifNotDryRun {
+            adminClient.createTopic(topic.name, topic.partitions, topic.replicationFactor, topic.config)
+          }
+        } yield {
+          // combining the 2 KafkaOperationResults' logs (they have no values)
+          deleteTopicsCheckResult.flatMap(_ => createTopicResult)

```

Comment:
Would it make more sense to have ths in the for com? asking as you're using flatMap here